daily.sh

```diff
@@ -126,19 +130,21 @@ set_notifiable_result() {
 #   Exit-Code: 0 >= min ver, 1 < min ver
 #######################################
 check_dependencies() {
-    local branch=$(git rev-parse --abbrev-ref HEAD)
+    local branch ver_56 ver_71 ver_72 python3 python_deps phpver pythonver old_branches msg
+
+    branch=$(git rev-parse --abbrev-ref HEAD)
     scripts/check_requirements.py > /dev/null 2>&1 || pip3 install -r requirements.txt > /dev/null 2>&1
 
-    local ver_56=$(php -r "echo (int)version_compare(PHP_VERSION, '5.6.4', '<');")
-    local ver_71=$(php -r "echo (int)version_compare(PHP_VERSION, '7.1.3', '<');")
-    local ver_72=$(php -r "echo (int)version_compare(PHP_VERSION, '7.2.5', '<');")
-    local python3=$(python3 -c "import sys;print(int(sys.version_info < (3, 4)))" 2> /dev/null)
-    local python_deps=$("${LIBRENMS_DIR}/scripts/check_requirements.py" > /dev/null 2>&1; echo $?)
-    local phpver="master"
-    local pythonver="master"
+    ver_56=$(php -r "echo (int)version_compare(PHP_VERSION, '5.6.4', '<');")
+    ver_71=$(php -r "echo (int)version_compare(PHP_VERSION, '7.1.3', '<');")
+    ver_72=$(php -r "echo (int)version_compare(PHP_VERSION, '7.2.5', '<');")
+    python3=$(python3 -c "import sys;print(int(sys.version_info < (3, 4)))" 2> /dev/null)
+    python_deps=$("${LIBRENMS_DIR}/scripts/check_requirements.py" > /dev/null 2>&1; echo $?)
+    phpver="master"
+    pythonver="master"
 
-    local old_branches="php53 php56 php71-python2"
-    if [[ " $old_branches " =~ " $branch " ]] && [[ "$ver_72" == "0" && "$python3" == "0" && "$python_deps" == "0" ]]; then
+    old_branches="php53 php56 php71-python2"
+    if [[ $old_branches =~ $branch ]] && [[ "$ver_72" == "0" && "$python3" == "0" && "$python_deps" == "0" ]]; then
```

Comment:
This is wrong

Reply:
care to explain why? AFAICT there is no functional change.

Reply:
Because it matches substrings delimited by spaces.
For example; ` php53 php56 php71-python2 ` =~ ` php56 `
Note the spaces.

To get shellcheck to be quiet you can change to == and use globbing.
Maybe;
```suggestion
    if [[ " $old_branches " == *" $branch "* ]] && [[ "$ver_72" == "0" && "$python3" == "0" && "$python_deps" == "0" ]]; then
```

Reply:
Else for example `php5` would match...

Reply:
so basically the check is reversed and the solution is adding spaces?

proper solution IMHO would be

```
old_branches="^(php53|php56|php71-python2)$"
if [[ $branch =~ $old_branches ]] && [[ "$ver_72" == "0" && "$python3" == "0" && "$python_deps" == "0" ]]; then
```

i.e. compare $branch to a regex of $old_branches

Reply:
@Jellyfrog can you please review?